[Bug] [JDBC SQLite] SQL error or missing database (near "VALUES": syntax error)

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

```sql
INSERT INTO main.`xxx` (`id`, `type`, `amount`, `ts`) VALUES (?, ?, ?, ?) ON CONFLICT(`id`,`type`) DO UPDATE SET `id`=VALUES(`id`), `type`=VALUES(`type`), `amount`=VALUES(`amount`), `ts`=VALUES(`ts`)
```

It seems that SQLite 3.x don't support VALUES(`xxx`) syntax in DO UPDATE section of UPSERT.

https://www.sqlite.org/lang_upsert.html
https://github.com/apache/seatunnel/blob/0fe175c43365695c13c23a020f41c45f8dfcbe00/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/sqlite/SqliteDialect.java#L51

### SeaTunnel Version

2.3.13

### SeaTunnel Config

```conf
env {
  parallelism = 1
  job.mode = "STREAMING"
  checkpoint.interval = 300000
}

source {
  Postgres-CDC {
    plugin_output = "pg_cdc"
    url = "jdbc:postgresql://postgres:5432/test"
    username = "${POSTGRES_USER}"
    password = "${POSTGRES_PASSWORD}"
    schema-names = ["public"]
    database-names = ["test"]
    table-names = [
      "test.public.xxx"
    ]
    slot.name = "pg2sqlite"
  }
}

sink {
  Jdbc {
    plugin_input = "pg_cdc"
    url = "jdbc:sqlite:/data/sqlite/data.db"
    driver = "org.sqlite.JDBC"
    table = "${table_name}"
    database = "main"
    generate_sink_sql = true
  }
}
```

### Running Command

```shell
command:
      - ./bin/seatunnel.sh
      - -m
      - local
      - -c
      - /data/jobs/pg2sqlite.seatunnel.conf
      - -i
      - POSTGRES_USER=${POSTGRES_USER:-postgres},POSTGRES_PASSWORD=${POSTGRES_PASSWORD:-postgres}
```

### Error Exception

```log
Caused by: org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (near "VALUES": syntax error)

	at org.sqlite.core.DB.newSQLException(DB.java:1135) ~[sqlite-jdbc-3.39.3.0.jar:?]

	at org.sqlite.core.DB.newSQLException(DB.java:1146) ~[sqlite-jdbc-3.39.3.0.jar:?]

	at org.sqlite.core.DB.throwex(DB.java:1106) ~[sqlite-jdbc-3.39.3.0.jar:?]
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/sqlite/SqliteDialect.java
+++ b/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/sqlite/SqliteDialect.java
@@ -55,9 +55,8 @@ public class SqliteDialect implements JdbcDialect {
                         .map(
                                 fieldName ->
                                         quoteIdentifier(fieldName)
-                                                + "=VALUES("
-                                                + quoteIdentifier(fieldName)
-                                                + ")")
+                                                + "=EXCLUDED."
+                                                + quoteIdentifier(fieldName))
                         .collect(Collectors.joining(", "));
 
         String conflictFields =

--- a/seatunnel-connectors-v2/connector-jdbc/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/sqlite/SqliteDialectTest.java
+++ b/seatunnel-connectors-v2/connector-jdbc/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/sqlite/SqliteDialectTest.java
@@ -1,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.jdbc.internal.dialect.sqlite;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class SqliteDialectTest {
+    @Test
+    void testUpsertStatement() {
+        SqliteDialect dialect = new SqliteDialect();
+        final String database = "seatunnel";
+        final String tableName = "role";
+        final String[] fieldNames = {
+            "id", "type", "role_name", "description", "create_time", "update_time"
+        };
+        final String[] uniqueKeyFields = {"id"};
+
+        String upsertSql =
+                dialect.getUpsertStatement(database, tableName, fieldNames, uniqueKeyFields)
+                        .orElseThrow(
+                                () ->
+                                        new AssertionError(
+                                                "Expected upsertSql String to be present"));
+        // SqliteDialect uses backticks (`) for quoting identifiers (table only, not database)
+        // Note: SqliteDialect always generates DO UPDATE SET, not DO NOTHING
+        Assertions.assertEquals(
+                "INSERT INTO seatunnel.`role` (`id`, `type`, `role_name`, `description`, `create_time`, `update_time`) VALUES (:id, :type, :role_name, :description, :create_time, :update_time) ON CONFLICT(`id`) DO UPDATE SET `id`=EXCLUDED.`id`, `type`=EXCLUDED.`type`, `role_name`=EXCLUDED.`role_name`, `description`=EXCLUDED.`description`, `create_time`=EXCLUDED.`create_time`, `update_time`=EXCLUDED.`update_time`",
+                upsertSql);
+    }
+}

